src/commands/cmd_query.c

```diff
@@ -61,12 +61,20 @@ void Graph_Query(void *args) {
 
 	QueryCtx_BeginTimer(); // Start query timing.
 
+	cypher_parse_result_t *params_parse_result = parse_params(command_ctx->query);
+	if(params_parse_result == NULL) goto cleanup;
+	if(AST_Validate_QueryParams(ctx, params_parse_result) != AST_VALID) goto cleanup;
+
+	// Query caching enabler.
+	AST_Extract_Params(params_parse_result);
```

Comment:
These steps should be unnecessary if a query doesn't have parameters, right? Why don't we retrieve the query statement earlier so we can put this in a conditional on `noptions > 0`?

Reply:
since the first thing this function does is 
```
// Retrieve the AST root node from a parsed query.
	const cypher_astnode_t *statement = cypher_parse_result_get_root(parse_result, 0);
	// We are parsing with the CYPHER_PARSE_ONLY_STATEMENTS flag,
	// and double-checking this in AST validations
	assert(cypher_astnode_type(statement) == CYPHER_AST_STATEMENT);
	uint noptions = cypher_ast_statement_noptions(statement);
	if(noptions == 0) return;
```
I can call it from within the validation since we alread have the statment there but it will break abstraction.
thoughts?

Reply:
I agree with your logic, this is fine as it is!